### PR TITLE
Fix wrong user returned in API

### LINKED ISSUE #15127: Wrong user returned in API: GET /repos/{owner}/{repo}/pulls/{index}/reviews/{id}/comments
- Gitea version (or commit ref): v1.13.5
- Git version: 2.29.2
- Operating system: Debian/buster
- Release from https://dl.gitea.io/gitea/1.13.5/gitea-1.13.5-linux-amd64
- Also tried self-built v1.13.5
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [X] MySQL
  - [ ] MSSQL
  - [ ] SQLite

## Description

I use the API to download the comments of a review.
In the returned json all comments have the the same user, although the comments were done by different users (as can be seen in the web frontend).

In modules/convert/pull_review.go ToPullReviewCommentList uses the reviewer, but I think it should use the poster of the comment. This fixes it for me:

    --- a/modules/convert/pull_review.go
    +++ b/modules/convert/pull_review.go
    @@ -94,7 +94,7 @@ func ToPullReviewCommentList(review *models.Review, doer *models.User) ([]*api.P
                    apiComment := &api.PullReviewComment{
                        ID:           comment.ID,
                        Body:         comment.Content,
    -                   Reviewer:     ToUser(review.Reviewer, doer != nil, auth),
    +                   Reviewer:     ToUser(comment.Poster, doer != nil, auth),
                        ReviewID:     review.ID,
                        Created:      comment.CreatedUnix.AsTime(),
                        Updated:      comment.UpdatedUnix.AsTime(),

### PR DESCRIPTION
**Pull request by sotho**
The API call: GET /repos/{owner}/{repo}/pulls/{index}/reviews/{id}/comments
returns always the reviewer, but should return the poster.

close #15127

### CODE REVIEW

File: modules/convert/pull_review.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func ToPullReviewCommentList(review *models.Review, doer *models.User) ([]*api.P
 				apiComment := &api.PullReviewComment{
 					ID:           comment.ID,
 					Body:         comment.Content,
-					Reviewer:     ToUser(review.Reviewer, doer != nil, auth),
+					Reviewer:     ToUser(comment.Poster, doer != nil, auth),
```
Comment (6543) [inline modules/convert/pull_review.go:98]:
```suggestion
					Reviewer:     ToUser(comment.Poster, doer != nil, doer.IsAdmin || doer.ID == comment.Poster.ID),
```

File: modules/convert/pull_review.go [hunk 2]
```diff
@@ -96,7 +96,7 @@ func ToPullReviewCommentList(review *models.Review, doer *models.User) ([]*api.P
...
 					ReviewID:     review.ID,
```
Comment (6543) [inline modules/convert/pull_review.go]:
```suggestion
					ReviewID:     comment.Poster.ID,
```

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
thanks for the pull :) @sotho  - finished it for you

File: modules/convert/pull_review.go [hunk 3]
```diff
@@ -85,19 +85,18 @@ func ToPullReviewCommentList(review *models.Review, doer *models.User) ([]*api.P
...
+				if doer != nil {
+					auth = doer.IsAdmin || doer.ID == comment.Poster.ID
+				}
 				apiComment := &api.PullReviewComment{
 					ID:           comment.ID,
 					Body:         comment.Content,
-					Reviewer:     ToUser(review.Reviewer, doer != nil, auth),
-					ReviewID:     review.ID,
+					Reviewer:     ToUser(comment.Poster, doer != nil, auth),
+					ReviewID:     comment.Poster.ID,
```
Comment (zeripath) [inline modules/convert/pull_review.go]:
```suggestion
					ReviewID:     comment.PosterID,
```

**Review by zeripath — APPROVED** [review]
Otherwise LGTM

**Comment by 6543** [comment]
:rocket:

**Comment by 6543** [comment]
@sotho pleace backport ... if you dont know just ask :)

**Comment by sotho** [comment]
Thanks. I cherry-picked the commit and opened two pull requests: for 1.13 and 1.14.

**Comment by sotho** [comment]
Hmm. I tried it out, but I think the change you added is not correct.
The ReviewID should stay as review.ID and not become comment.Poster.ID.

The struct is misleading: the "Reviewer" should be "Commenter" or "Poster", something like:

	Commenter:  ToUser(comment.Poster, doer != nil, auth),
	ReviewID:   review.ID,

**Comment by zeripath** [comment]
Damn I think @sotho is right here. it should be reviewID

**Comment by 6543** [comment]
-> #15164

and after that adjust backports + merge

**Pull request merged**